specs/data_structures.md

```diff
@@ -66,16 +72,16 @@ Blocks are the top-level data structure of the LazyLedger blockchain.
 
 Block header, which is fully downloaded by both full clients and light clients.
 
-| name                | type                      | description                                                                                  |
-| ------------------- | ------------------------- | -------------------------------------------------------------------------------------------- |
-| `height`            | `uint64`                  | Block height. The genesis block is at height `1`.                                            |
-| `time`              | [Time](#time)             | Timestamp of this block.                                                                     |
-| `lastBlockID`       | [BlockID](#blockid)       | Previous block's ID.                                                                         |
-| `lastCommitRoot`    | [HashDigest](#hashdigest) | Previous block's Tendermint commit root.                                                     |
-| `consensusRoot`     | [HashDigest](#hashdigest) | Merkle root of [consensus parameters](#consensus-parameters) for this block.                 |
-| `stateCommitment`   | [HashDigest](#hashdigest) | Commitment to state root and validator set root after this block's transactions are applied. |
-| `availableDataRoot` | [HashDigest](#hashdigest) | Root of [commitments to erasure-coded data](#availabledataheader).                           |
-| `proposerAddress`   | [Address](#address)       | Address of this block's proposer.                                                            |
+| name                | type                      | description                                                                  |
+| ------------------- | ------------------------- | ---------------------------------------------------------------------------- |
+| `height`            | `uint64`                  | Block height. The genesis block is at height `1`.                            |
+| `time`              | [Time](#time)             | Timestamp of this block.                                                     |
+| `lastBlockID`       | [BlockID](#blockid)       | Previous block's ID.                                                         |
+| `lastCommitRoot`    | [HashDigest](#hashdigest) | Previous block's Tendermint commit root.                                     |
+| `consensusRoot`     | [HashDigest](#hashdigest) | Merkle root of [consensus parameters](#consensus-parameters) for this block. |
+| `stateCommitment`   | [HashDigest](#hashdigest) | The [state root](#state) after this block's transactions are applied.        |
```

Comment:
If it is clear that we will stick with deferred execution for a while (probably after launch even), should the first version of the spec adhere to this too? (then this would be state after the last block's transactions are applied).

Whenever (if) we decide to drop deferred execution, we can update the spec to another version.

Reply:
We need immediate execution for fee burning (which IMO is 100% necessary, for testnet at the latest). Minting new coins is a source, and there are two ways to have sinks:
1. state rent
1. fee burning

We can't really do the former for obvious reasons, so we're left with the latter. Sinks create intrinsic demand for the currency, a property that cannot be accomplished any way else (which is why taxes---a sink---are still a thing even if the government can print money---a source). And fee burning requires immediate execution.

Reply:
I was under the impression, we will have a first test-net without switching to immediate execution. If we will use deferred execution only in dev-nets, it's probably OK that the spec and the implementation diverge for a while I guess. 

We might have to re-discuss this when we are closer to a first dev-net iteration.

Reply:
By testnet above I guess I meant "testnet with mainnet configuration." We could have more public testnets (e.g. PoA ones) before that, along with non-feature-complete devnets, that don't need immediate execution because there are no incentives and we can just assume the single operator isn't making invalid blocks.